packages/venia-concept/src/components/CreateAccount/createAccount.js

```diff
@@ -0,0 +1,220 @@
+import { createElement, Component } from 'react';
+import PropTypes from 'prop-types';
+import Input from 'src/components/Input';
+import { HelpTypes } from 'src/components/Input';
+import Button from 'src/components/Button';
+import defaultClasses from './createAccount.css';
+import classify from 'src/classify';
+import Form from 'src/components/Form';
+import { debounce } from 'underscore';
+import { RestApi } from '@magento/peregrine';
+import ErrorDisplay from 'src/components/ErrorDisplay';
+import Checkbox from 'src/components/Checkbox';
+
+const { request } = RestApi.Magento2;
+
+class CreateAccount extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            root: PropTypes.string,
+            createAccountError: PropTypes.string
+        }),
+
+        createAccountError: PropTypes.object,
+        createAccount: PropTypes.func
+    };
+
+    state = {
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        passwordConfirm: '',
+        subscribe: false,
+        checkingEmail: false,
+        emailAvailable: false,
+        subscribe: false
+    };
+
+    get errorMessage() {
+        const { createAccountError } = this.props;
+        return <ErrorDisplay error={createAccountError} />;
+    }
+
+    get hasEmailError() {
+        return (
+            !!this.state.email &&
+            !this.state.emailAvailable &&
+            !this.state.checkingEmail
+        );
+    }
+
+    get passwordConfirmError() {
+        return this.state.password !== this.state.passwordConfirm;
+    }
+
+    get disableAccountCreation() {
+        return (
+            this.hasEmailError ||
+            this.passwordConfirmError ||
+            !this.state.email ||
+            !this.state.firstName ||
+            !this.state.lastName
+        );
+    }
+
+    get emailHelpText() {
+        return this.hasEmailError
+            ? 'This email is already in use'
+            : 'Use an active email';
+    }
+
+    get emailHelpType() {
+        return this.hasEmailError ? HelpTypes.error : HelpTypes.hint;
+    }
+
+    get passwordHelpText() {
+        return this.passwordConfirmError ? 'Passwords do not match' : '';
+    }
+
+    get passwordHelpType() {
+        return HelpTypes.error;
+    }
+
+    render() {
+        const { classes } = this.props;
+        const {
```

Comment:
I like this pattern. You've done all the validation right here, through destructuring some getters. Nicely done.

People newer to JS might not understand what's happening here; it would be good to add a comment.